src/widget/wtracktableview.cpp

```diff
@@ -1398,6 +1404,31 @@ void WTrackTableView::slotImportTrackMetadata() {
     }
 }
 
+void WTrackTableView::slotExportTrackMetadata() {
+    if (!modelHasCapabilities(TrackModel::TRACKMODELCAPS_IMPORTMETADATA)) {
+        return;
+    }
+
+    QModelIndexList indices = selectionModel()->selectedRows();
+
+    TrackModel* trackModel = getTrackModel();
+
+    if (trackModel == NULL) {
+        return;
+    }
+
+    foreach (QModelIndex index, indices) {
+        TrackPointer pTrack = trackModel->getTrack(index);
+        if (pTrack) {
+            // Export of metadata is deferred until all references to the
+            // corresponding track object have been dropped. Otherwise
+            // writing to files that are still used for playback might
+            // cause crashes or at least audible glitches!
```

Comment:
My use case for "Export Metadata into File" would be, to have the changes almost instantly in the file to be able to verify this with a third party tool. 
When is this not the case? Should we warn the user? There might be race conditions with the third party tool since the open file handle form the other tool will point to the old deleted file.

Reply:
Concurrent writes from external programs are out of scope. This can only be prevented by the OS.

I agree that we should display a dialog with an informational message that the actual export is deferred until Mixxx decides that writing metadata is safe. In the worst case this might not happen before Mixxx is shut down.

Reply:
Maybe this could be a one time warning with a "do not warn again" checkbox.

Reply:
Just noticed you have it already. I will test how this feels.